File: src/test/java/com/raychenon/leetcode/SortArrayByParityTest.java
package com.raychenon.leetcode;

import org.junit.Assert;
import org.junit.BeforeClass;
import org.junit.Test;

import java.util.Arrays;

/**
 * User: raychenon
 * Date: 2019-10-19
 * https://leetcode.com/problems/sort-array-by-parity/
 */
public class SortArrayByParityTest {

    private static SortArrayByParity sortArrayByParity;

    @BeforeClass
    public static void setUp() {
        sortArrayByParity = new SortArrayByParity();
    }

    @Test
    public void example1Test() {
        int[] input = new int[]{3, 1, 2, 4};
        int[] output = new int[]{2, 4, 3, 1};
        Assert.assertTrue(Arrays.equals(output, sortArrayByParity.sortArrayByParityTwoPasses(input)));

        int[] outputInPlace = new int[]{4,2,1,3};
        Assert.assertTrue(Arrays.equals(outputInPlace, sortArrayByParity.sortArrayByParityInPlace(input)));
    }
}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/SortArrayByParityTest.kt
package com.raychenon.kotlin.leetcode

import org.junit.Assert
import org.junit.Test
import java.util.*

/**
 * User: raychenon
 * Date: 2019-10-19
 * https://leetcode.com/problems/sort-array-by-parity/
 */
class SortArrayByParityTest {

    @Test
    fun example1Test() {
        Assert.assertTrue(
            Arrays.equals(
                intArrayOf(2, 4, 3, 1),
                SortArrayByParity.sortArrayByParity(intArrayOf(3, 1, 2, 4))
            )
        )

    }

    @Test
    fun example1ConstantSpaceTest() {
        Assert.assertTrue(
            Arrays.equals(
                intArrayOf(4, 2, 1, 3),
                SortArrayByParity.sortArrayByParityInPlace(intArrayOf(3, 1, 2, 4))
            )
        )
    }

}